docs/upload-strategies/transloadit.mdx

```diff
@@ -0,0 +1,585 @@
+      fields: {
+        message: file.meta.message,
+      },
+    }
+  },
+})
+```
+
+</details>
+
+#### `waitForEncoding`
+
+Wait for the template to finish, rather than only the upload, before marking the upload complete (`Boolean`, default: `false`).
+
+*   When `false`, rhe Assemblies will complete (or error) in the background but Uppy won’t know or care about it.
+*   When `true`, the Transloadit plugin waits for Assemblies to complete before the files are marked as completed.
+    This means users have to wait for a potentially long time, depending on how complicated your Assembly instructions are.
+    But, you can receive transcoding results on the client side, and have a fully client-side experience this way.
+
+When this is enabled, you can listen for the [`transloadit:result`](#transloadit-result) and [`transloadit:complete`](#transloadit-complete) events.
+
+#### `waitForMetadata`
+
+Wait for for Transloadit’s backend to catch early errors, not the entire template. (`Boolean`, default: `false`)
+
+When set to `true`, the Transloadit plugin waits for Transloadit’s backend to extract metadata from all the uploaded files.
+This is mostly handy if you want to have a quick user experience (so your users don’t necessarily need to wait for all the encoding to complete),
+but you do want to let users know about some types of errors that can be caught early on, like file format issues.
+
+You you can listen for the [`transloadit:upload`](#transloadit-upload) event when it’s enabled.
+
+#### `importFromUploadURLs`
+
+Allow another plugin to upload files, and then import those files into the Transloadit Assembly (`Boolean`, default: `false`).
+
+When enabling this option, Transloadit will _not_ configure the Tus plugin to upload to Transloadit.
+Instead, a separate upload plugin must be used.
+Once the upload completes, the Transloadit plugin adds the uploaded file to the Assembly.
+
+For example, to upload files to an S3 bucket and then transcode them:
+
+```js
+uppy.use(AwsS3, {
+  getUploadParameters (file) {
+    return { /* upload parameters */ }
+  },
+})
+uppy.use(Transloadit, {
+  importFromUploadURLs: true,
+  assemblyOptions: {
+    params: {
+      auth: { key: 'YOUR_API_KEY' },
+      template_id: 'YOUR_TEMPLATE_ID',
+    }
+  },
+})
+```
+
+:::note
+For this to work, the upload plugin must assign a publically accessible `uploadURL` property to the uploaded file object.
+The Tus and S3 plugins both do this automatically.
+For the XHRUpload plugin, you may have to specify a custom `getResponseData` function.
+:::
+
+#### `alwaysRunAssembly`
+
+Always create and run an Assembly when `uppy.upload()` is called, even if no files were selected (`Boolean`, default: `false`).
+
+This allows running Assemblies that do not receive files,
+but instead use a robot like [`/s3/import`](https://transloadit.com/docs/transcoding/#s3-import) to download
+the files from elsewhere, for example, for a bulk transcoding job.
+
+#### `locale`
+
```

Comment:
This could use an introductory sentence maybe

Reply:
This is a bigger todo, as this section will be generated.